### PR TITLE
Add oe_verify_report_internal() to liboehostverify.a

### LINKED ISSUE #3312: hostverify library: Cannot use oe_verify_attestation_certificate_with_evidence
This is with current `master`:
```
../openenclave/lib/openenclave/host/liboehostverify.a(verifier.c.o): In function `_verify_report':
.../common/sgx/verifier.c:742: undefined reference to `oe_verify_report_internal'
```
Looks like some sources are missing in the archive.

### PR DESCRIPTION
**Pull request by shants**
fixes #3312

Signed-off-by: Shantanu Sharma <shanshar@microsoft.com>

### CODE REVIEW

**Comment by qiucwang** [comment]
bors try

File: host/sgx/report.c [hunk 1]
```diff
@@ -130,6 +133,7 @@ oe_result_t oe_get_report_v2(
     if (!enclave || !report_buffer || !report_buffer_size)
         OE_RAISE(OE_INVALID_PARAMETER);
 
+#ifndef OE_BUILD_HOST_VERIFY
```
Comment (yentsanglee) [inline host/sgx/report.c:136]: Host_verify package never call oe_get_report*(). We can remove the entire function if possible.

File: host/sgx/report.c [hunk 2]
```diff
@@ -249,13 +262,13 @@ oe_result_t oe_verify_report_internal(
     }
     else if (header->report_type == OE_REPORT_TYPE_SGX_LOCAL)
     {
-        oe_result_t retval;
-
         if (enclave == NULL)
             OE_RAISE(OE_INVALID_PARAMETER);
-
+#ifndef OE_BUILD_HOST_VERIFY
```
Comment (yentsanglee) [inline host/sgx/report.c:267]: Passing report_type OE_REPORT_TYPE_SGX_LOCAL to host_verify package should be treated as a error. We can exclude this `else if` block and let `else` handle it.

File: host/sgx/report.c (hunk 1 above)
Comment (shants) [reply host/sgx/report.c]: host/sgx/report.c now has oe_get_report_v2(), moved rest to  host/sgx/report_common.c.

File: host/sgx/report.c (hunk 2 above)
Comment (shants) [reply host/sgx/report.c]: Done.

**Comment by yentsanglee** [comment]
bors try

**Pull request merged**